### ISSUE TITLE
if auto-redirect despite WCAG but UA blocks, tell user where to go 

### ISSUE THREAD

**Issue by joshueoconnor** [w3c/wcag#102]
https://lists.w3.org/Archives/Public/public-comments-wcag20/2015Jun/0002.html 

Name: Nick Levinson
Email: Nick_Levinson@yahoo.com
Affiliation: none
Document: W2
Item Number: Guideline 2.2: Provide users with disabilities enough time to read...
Part of Item: 
Comment Type: technical
Summary of Issue: if auto-redirect despite WCAG but UA blocks, tell user where to go
Comment (Including rationale for any proposed change):
Pages that refresh so that information is lost may violate WCAG 2.0 guideline 2.2, but refreshing is common anyway. However, a browser may prevent refreshing unless the user chooses to refresh. That's a good solution, but with one caveat: the user may not know why refreshing is supposed to occur, and the user should be well-informed so they can decide what to do next, which may be to do what the redirection was supposed to do. Therefore, the page before refreshing should say why it should refresh.

Example: I recently designed a website with a complicated directory structure. Many directories contained only subdirectories. A visitor to one would get a message from my host saying basically that there's nothing to see here. So, I prepared nonroot index.html files that would auto-redirect the visitor to a page. (I since changed my mind, and they no longer auto-refresh.) In the time before auto-redirection, the page would say what would happen, and would provide links, one to do the same thing as auto-redirection and the other to provide an alternative destination (the home page). So, if the browser prevented redirection, the user would know what to do anyway.

Proposed Change:
If auto-redirection is used despite violating WCAG, since a browser may block auto-redirection, tell the user what the intent was and provide a means of doing it manually. This advice can go into a document that supports WCAG today or a future revision of WCAG.

**Comment by MikeElledge** [comment]
The situation occurs when an auto-redirect is prevented by a browser without an explanation or link to the destination page.  The proposed solution is to provide users with an explanation and a link if an auto-redirect is prevented by a browser so they will 1) know why a page is auto-directing, and 2) decide if they want to continue the auto-direct process. I don't know if Nick was looking for a technique or failure for the issue, but they thought it was covered by Guideline 2.2 Enough Time.

It seems to me that the issue is that the user needs to be informed of an unexpected change in context (Guideline 3.2) before initiating an action (such as clicking on a link), and that having sufficient time to respond is a different concern. My suggestion is to include the auto-direct use case as a Failure when users are 1) not informed they are being re-directed and/or 2) they are not given an explanation for the re-direct.

**Comment by DavidMacDonald** [comment]
It seems to affect everyone equally, a blind person and a sighted person wouldn't have access to the new url. 

I think it is a browser issue...

direct mapping to 3.2.1, 3.2.2 or 2.2.2 are precarious I'd say... change of context is upon focus or input... there is no movement on the page so 2.2. is also precarious...

**Comment by MikeElledge** [comment]
Hi David--

I don't disagree with it being a usability issue for everyone, but it seems to me that it would have a greater impact on someone who is blind or has cognitive issues.

In reading 3.2.2, though, I'm surprised that giving users advance notice of what will happen when they click on a link (or tab) is only an advisory technique, and not required for conformance. Do you recall why prior notification isn't a requirement?

Mike

**Comment by Ryladog** [comment]
I agree that this issue is greater for persons with disabilities………

 

 

 

* katie *

 

Katie Haritos-Shea 
Senior Accessibility SME (WCAG/Section 508/ADA/AODA)

 

Cell: 703-371-5545 |  <mailto:ryladog@gmail.com> ryladog@gmail.com | Oakton, VA |  <http://www.linkedin.com/in/katieharitosshea/> LinkedIn Profile | Office: 703-371-5545

**Comment by DavidMacDonald** [comment]
HI Mike
RE your question about spawned windows

WCAG 1 required notification  of a new window.

10.1 Until user agents allow users to turn off spawned windows, do not cause pop-ups or other windows to appear and do not change the current window without informing the user. [Priority 2]

By the time WCAG 2 was being developed screen readers notified users that it was a new window, and a simple keystroke closed the window. Usually CTL-W placing the user back where they came from. Some on the group argued that this was sufficient, others felt notification was necessary still. I was actually one of those arguing in favour of notifying users of new windows. But the consensus turned towards not requiring notification, because it was argued that it did not disadvantage people with disabilities any more than users who were sighted, and had not disabilities.

I think it is important to honour consensus, even if it's something I don't agree with, that's what gave WCAG it's authority and made it respected, so I try to always teach the WCAG as it is written, not as I wish it was on a particular success criteria.

**Comment by DavidMacDonald** [comment]
HI Katie and Mike

When you say greater impact perhaps, do you mean that a person who is blind or has cognitive disabilities couldn't as easily look a code view or do a google search for the direct link as well as a sighted person or someone with "normal" cognitive facilities...

For me there is no link there and it didn't forward, it seems like a brick wall to all normal users who don't use hacks to find the location. 

But even if it IS harder, I still have trouble mapping it... We could map to guideline 3.2, but I don't think a SC for reasons mentioned above... that would seem like a shoehorning a good failure into a SC to which it doesn't really apply. Someone might eventually call us on it, even though we all agree it is bad behaviour for a browser.

**Comment by DavidMacDonald** [comment]
I think we could punt this to the User Agent guidelines which are still being worked on, we could tell them what we think it does and why it is important. they have the power to change things.

**Comment by Ryladog** [comment]
On the other hand they will be closing that group and its members will be joining ours relatively soon so….

 

 

 

* katie *

 

Katie Haritos-Shea 
Senior Accessibility SME (WCAG/Section 508/ADA/AODA)

 

Cell: 703-371-5545 |  <mailto:ryladog@gmail.com> ryladog@gmail.com | Oakton, VA |  <http://www.linkedin.com/in/katieharitosshea/> LinkedIn Profile | Office: 703-371-5545

**Comment by lauracarlson** [comment]
From the [Cognitive Accessibility User Research](http://www.w3.org/TR/coga-user-research) document: [3.7.6.6 Orientation and Error Prevention/Recovery](http://www.w3.org/TR/coga-user-research/#h-orientation-and-error-prevention-recovery)

> Give users control over time sensitive content changes: Avoid automatic refreshes or redirects. Allow users to control content updates or changes. Avoid unnecessary time-outs or expirations. Allow users to request more time.

But I haven't found a [COGA technique](https://rawgit.com/w3c/coga/master/techniques/index.html) that addresses it.

**Comment by MikeElledge** [comment]
Thanks for the good comments, everyone. I'm leaning toward David's POV that it is a usability issue, since it would be difficult at best for anyone to know where the redirect would take them, and is not an accessibility issue per se. If we accept that idea (and please respond if you disagree), then how do you feel about the following comment for the User Agent team: 

"Give users the option of accepting or declining an auto-direct on a site, either as a browser preference or at the moment of auto-direct. At the moment of auto-direct inform them of the destination page address and give them the option of proceeding or returning/remaining on the current page."

Thoughts?

**Comment by MikeElledge** [comment]
Thanks for the good comments, everyone. 

I'm leaning toward David's POV that it is a usability issue, since it would be difficult at best for anyone to know where the redirect would take them, and is not an accessibility issue per se. If we accept that idea (and please respond if you disagree), then how do you feel about the following comment for the User Agent team: 
"Give users the option of accepting or declining an auto-direct on a site, either as a browser preference or at the moment of auto-direct. At the moment of auto-direct inform them of the destination page address and give them the option of proceeding or returning/remaining on the current page."
Thoughts?
Mike
   


     On Wednesday, July 22, 2015 9:52 AM, Laura Carlson <notifications@github.com> wrote:
   

 From the Cognitive Accessibility User Research document: 3.7.6.6 Orientation and Error Prevention/Recovery
Give users control over time sensitive content changes: Avoid automatic refreshes or redirects. Allow users to control content updates or changes. Avoid unnecessary time-outs or expirations. Allow users to request more time.
But I haven't found a COGA technique that addresses it.—

**Comment by joshueoconnor** [comment]
Working group survey to comment and close out this item: https://www.w3.org/2002/09/wbs/35422/Sept1st2015/

**Comment by joshueoconnor** [comment]
RESOLUTION: Leave open, more info needed. Discuss with Mike to see if we can close.
No definitive accessibility related use case, though it is a usability fail/issue.

**Comment by joshueoconnor** [comment]
@MikeElledge Will you be able to make the call on Tues 15th to discuss? Please review minutes if possible, the consensus was this may not have a definitive a11y related case.  Your input on the call would be helpful. Thanks

**Comment by mraccess77** [comment]
This seems like a user agent issue that is best addressed outside of WCAG.

**Comment by MikeElledge** [comment]
Thanks, Jonathan.
Mike
 


    On Tuesday, December 1, 2015 1:44 PM, Jonathan Avila <notifications@github.com> wrote:
 

 This seems like a user agent issue that is best addressed outside of WCAG.—

**Comment by awkawk** [comment]
I do believe that this is an issue but don't think that it is more likely to be an issue for people with disabilities as any user might configure their browser to prevent such redirects.  As for when it occurs I don't think that a disabled user is any more lost than any other user if both are not provided any guidance around where the redirect would have taken them.

I think that we should write a response to the commenter that indicates that this is an issue but that it is not covered by WCAG as it is a usability issue that affects all users.

**Comment by MikeElledge** [comment]
I agree with Andrew's comment. Does anyone have an objection? Otherwise I will respond to the question that this is a general usability issue and therefore not subject to WCAG 2.0.
Mike 


    On Tuesday, December 1, 2015 5:15 PM, Andrew Kirkpatrick <notifications@github.com> wrote:
 

 I do believe that this is an issue but don't think that it is more likely to be an issue for people with disabilities as any user might configure their browser to prevent such redirects. As for when it occurs I don't think that a disabled user is any more lost than any other user if both are not provided any guidance around where the redirect would have taken them.I think that we should write a response to the commenter that indicates that this is an issue but that it is not covered by WCAG as it is a usability issue that affects all users.—

**Comment by awkawk** [comment]
Proposed response (From Mike):
Thank you for your comment. Your approach seems like a good approach to handle the issue you describe. Our consensus is that although it is a usability problem, it is not accessibility-specific so isn't covered by WCAG 2.0. Our reasoning is that this issue has a negative impact on all users, and not just persons with disabilities.

**Comment by MikeElledge** [comment]
Thanks, Andrew! Now I know how it works!

Mike
 


    On Thursday, December 3, 2015 5:15 PM, Andrew Kirkpatrick <notifications@github.com> wrote:
 

 Proposed response (From Mike):
Thank you for your comment. Your approach seems like a good approach to handle the issue you describe. Our consensus is that although it is a usability problem, it is not accessibility-specific so isn't covered by WCAG 2.0. Our reasoning is that this issue has a negative impact on all users, and not just persons with disabilities. —

**Comment by awkawk** [comment]
CfC accepted: https://lists.w3.org/Archives/Public/w3c-wai-gl/2015OctDec/0329.html

Responded to commenter: https://lists.w3.org/Archives/Public/public-comments-wcag20/2015Dec/0006.html
